bolt-simple-taproot.md

```diff
@@ -0,0 +1,1141 @@
+    using the `Sign` algorithm from `bip-musig2`.
+  
+  - MUST include the partial signature and the public counterpart of the
+    generated nonce in the `partial_signature_with_nonce` field.
+
+The recipient:
+
+  - MUST fail the channel if `partial_signature_with_nonce` is absent.
+  
+  - MUST compute the aggregate nonce from:
+  
+    - the `next_local_nonce` field the recipient previously sent in
+      `funding_created`
+  
+    - the `public_nonce` included as part of the `partial_signature_with_nonce`
+      field
+  
+  - MUST verify the `partial_signature_with_nonce` field using the
+    `PartialSigVerifyInternal` algorithm of `bip-musig2`:
+  
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `channel_ready` Extensions
+
+We add a new TLV field to the `channel_ready` message:
+
+1. `tlv_stream`: `channel_ready_tlvs`
+2. types:
+   1. type: 4 (`next_local_nonce`)
+   2. data:
+      * [`66*byte`: `public_nonce`]
+
+Similar to the `next_per_commitment_point`, by sending the `next_local_nonce`
+value in this message, we ensure that the remote party has our public nonce
+which is required to generate a new commitment signature.
+
+##### Requirements
+
+The sender:
+
+- MUST set `next_local_nonce` to a fresh, unique `musig2` nonce as specified by
+  `bip-musig2`
+
+The recipient:
+
+- MUST fail the channel if `next_local_nonce` is absent.
+
+### Cooperative Closure
+
+#### `shutdown` Extensions
+
+A new TLV field is added to the `shutdown` message:
+
+1. type: 38 (`shutdown`)
+2. data:
+    * ...
+    * [`shutdown_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `shutdown_tlvs`
+2. types:
+    1. type: 8 (`shutdown_nonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+
+Before a signature can be generated for a co-op close transaction, but sides
+must exchange a fresh pair of `musig2_pubnonce` values. We package this with
+the `shutdown` message so that both sides can send a `closing_signed` message
+as soon as a `shutdown` message is sent and received.
+
+Only a single nonce is needed as there's only a single message to sign: the
+shared co-op close transaction.
+
+##### Requirements
+
+A sending node:
+
+ - MUST set the `shutdown_nonce` to a valid `musig2` public nonce.
+
+A receiving node:
+
+ - MUST verify that the `shutdown_nonce` value is a valid `musig2` public
+   nonce.
+
+#### `closing_signed` Extensions
+
+We add a new TLV to the `closing_signed`
+message's existing `tlv_stream`:
+
+1. `tlv_stream`: `closing_signed_tlvs`
+2. types:
+   1. type: 6 (`partial_signature`)
+   2. data:
+      * [`32*byte`: `partial_signature`
+
+Both sides **MUST** provide this new TLV field.
+
+Once all partial signatures are known, the `PartialSigAgg` algorithm is used to
+aggregate the partial signature into a final, valid [BIP
+340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki) Schnorr
+signature.
+
+#### Requirements
+
+The sender:
+
+  - MUST set the original, non-TLV `signature` field to a 0-byte-array of length
```

Comment:
> why cant we drop the unused fields since these messages wont ever need to be parsed by nodes not compatible with this spec

Less changes, everyone is able to re-use all their parsing, and just read the new TLV fields. 

>  Further, why not change all messages in the funding flow after open_channel & accept_channel to be purely tlv?

Less changes, the messages are nearly entirely the same. All the existing fields have their same meaning, we just now add the nonce + sig fields.